File: clients/cpp/model/Vec2Float.cpp
#include "Vec2Float.hpp"

Vec2Float::Vec2Float() { }
Vec2Float::Vec2Float(float x, float y) : x(x), y(y) { }
Vec2Float Vec2Float::readFrom(InputStream& stream) {
    Vec2Float result;
    result.x = stream.readFloat();
    result.y = stream.readFloat();
    return result;
}
void Vec2Float::writeTo(OutputStream& stream) const {
    stream.write(x);
    stream.write(y);
}
std::string Vec2Float::toString() const {
    return std::string("Vec2Float") + "(" +
        std::to_string(x) +
        std::to_string(y) +
        ")";
}

File: clients/csharp/Model/Vec2Float.cs
namespace AiCup2019.Model
{
    public struct Vec2Float
    {
        public float X { get; set; }
        public float Y { get; set; }
        public Vec2Float(float x, float y)
        {
            this.X = x;
            this.Y = y;
        }
        public static Vec2Float ReadFrom(System.IO.BinaryReader reader)
        {
            var result = new Vec2Float();
            result.X = reader.ReadSingle();
            result.Y = reader.ReadSingle();
            return result;
        }
        public void WriteTo(System.IO.BinaryWriter writer)
        {
            writer.Write(X);
            writer.Write(Y);
        }
    }
}
